engine/combat/controllers/first_encounter_controller.py

```diff
@@ -18,7 +18,7 @@
 class FirstEncounterController(EncounterController):
     def generate_action(self: Self) -> bool:
```

Comment:
Not a required thing, but all of the custom encounter controllers have the exact same code for this function override. Could create a `CustomEncounterController` that overrides this function, and have `FirstEncounterController` etc inherit from that.

Not sure if worth it tho.